test/phoenix_ecto/plugs/check_repo_ready_test.exs

```diff
@@ -0,0 +1,209 @@
+
+    conn = conn(:get, "/")
+
+    assert conn ==
+             CheckRepoReady.call(conn,
+               otp_app: :check_repo_ready,
+               get_migrations_function: get_migrations_function
+             )
+  after
+    Application.delete_env(:check_repo_ready, :ecto_repos)
+    Process.unregister(StorageStatusRepo)
+  end
+
+  test "does not raise an error when using the fallback get_migration_function" do
+    Process.register(self(), StorageUpRepo)
+    Application.put_env(:check_repo_ready, :ecto_repos, [StorageUpRepo])
+
+    conn = conn(:get, "/")
+
+    assert conn == CheckRepoReady.call(conn, otp_app: :check_repo_ready)
+  after
+    Application.delete_env(:check_repo_ready, :ecto_repos)
+    Process.unregister(StorageUpRepo)
+  end
+
+  test "does not raise an error when storage_status returns {:error, term()}" do
+    Process.register(self(), StorageErrorRepo)
+    Application.put_env(:check_repo_ready, :ecto_repos, [StorageErrorRepo])
+    get_migrations_function = fn _repo -> [] end
+
+    conn = conn(:get, "/")
+
+    assert conn ==
+             CheckRepoReady.call(conn,
+               otp_app: :check_repo_ready,
+               get_migrations_function: get_migrations_function
+             )
+  after
+    Application.delete_env(:check_repo_ready, :ecto_repos)
+    Process.unregister(StorageErrorRepo)
+  end
+
+  test "does not raise an error when the storage is created and there are no pending migrations for multiple repos" do
+    Process.register(self(), StorageUpRepo)
+    Application.put_env(:check_repo_ready, :ecto_repos, [StorageUpRepo, StorageUpRepo])
+
+    get_migrations_function = fn _repo -> [] end
+
+    conn = conn(:get, "/")
+
+    assert conn ==
+             CheckRepoReady.call(
+               conn,
+               otp_app: :check_repo_ready,
+               get_migrations_function: get_migrations_function
+             )
+  after
+    Application.delete_env(:check_repo_ready, :ecto_repos)
+    Process.unregister(StorageUpRepo)
+  end
+
+  test "raises an error when one of multiple repos does not have the database created" do
+    Process.register(self(), StorageUpRepo)
+    Process.register(spawn(&LongLivedProcess.run/0), StorageDownRepo)
+
+    Application.put_env(:check_repo_ready, :ecto_repos, [StorageUpRepo, StorageDownRepo])
+
+    get_migrations_function = fn _repo -> [] end
+
+    conn = conn(:get, "/")
+
+    assert_raise(Phoenix.Ecto.StorageNotCreatedError, fn ->
+      CheckRepoReady.call(
+        conn,
+        otp_app: :check_repo_ready,
+        get_migrations_function: get_migrations_function
+      )
+    end)
+  after
+    Application.delete_env(:check_repo_ready, :ecto_repos)
+    Process.unregister(StorageUpRepo)
+    Process.unregister(StorageDownRepo)
+  end
+
+  test "raises an error when one of multiple repos has pending migrations" do
+    Process.register(self(), StorageUpRepo)
+    Process.register(spawn(&LongLivedProcess.run/0), NoStorageStatusRepo)
```

Comment:
```suggestion
    Process.register(spawn_link(&LongLivedProcess.run/0), NoStorageStatusRepo)
```

Otherwise the process never finishes. We need to do it in the others too.

Reply:
Nice!

[fixed here](https://github.com/phoenixframework/phoenix_ecto/compare/693fbe89c367841069104dcf2070e1e40f9f0028..d85122ccded004efe2b01f32d66e64073c16766f)